packages/squiggle-lang/src/rescript/Distributions/GenericDist/GenericDist.res

```diff
@@ -258,7 +258,26 @@ module AlgebraicCombination = {
         : Convolution(convOp)
     }
 
+  let chooseConvolutionOrMonteCarlo = (
+    ~strat: DistributionTypes.asAlgebraicCombinationStrategy,
+    op: Operation.algebraicOperation,
+    t2: t,
+    t1: t,
+  ): result<calculationMethod, error> => {
+    switch strat {
+    | AsDefault => Ok(chooseConvolutionOrMonteCarloDefault(op, t2, t1))
+    | AsConvolution =>
+      switch op {
+      | #Divide | #Power | #Logarithm => Error(RequestedStrategyInvalidError)
+      | (#Add | #Subtract | #Multiply) as convOp => Ok(Convolution(convOp))
+      }
+    | AsMontecarlo => Ok(MonteCarlo)
+    | AsSymbolic => Error(RequestedStrategyInvalidError)
+    }
+  }
+
   let run = (
```

Comment:
This is the function that needs to be changed. It starts off doing a symbolic manipulation. You can't wait until after.

<img width="1107" alt="image" src="https://user-images.githubusercontent.com/377065/165393536-bd3e59ae-1883-4290-b0e7-db0a5efb6288.png">